sql: add a note about `FULLTEXT` (#17876)
pingcap/docs: sql-statements/sql-statement-create-table.md

## MySQL compatibility

* All of the data types except `SPATIAL` types are supported.
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.

> * `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.

Reviewer: 243 and 244 are now saying the same. I like the second line more than the first.

Author: ok deleted the first one.

> * `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.

Author: ```suggestion
```

## MySQL compatibility

> * All of the data types except `SPATIAL` types are supported.

Reviewer 2: Spatial type means https://dev.mysql.com/doc/refman/8.4/en/spatial-type-overview.html i.e. `GEOMETRY`, `POLYGON`, `MULTIPOINT` etc. There isn't a single type called `SPATIAL`, and this item is not talking about spatial index either.

> * All of the data types except `SPATIAL` types are supported.

Author: ```suggestion
* All of the data types except spatial types are supported.
```

## MySQL compatibility

* All of the data types except spatial types are supported.

> * TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.

Reviewer 3: ```suggestion
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
```
